src/Controller/DefaultController.php

```diff
@@ -395,95 +396,154 @@ public function pickup_locations_for_patron() {
-        $link = $host . '/pickuprequest/' . $patronId . '/'. $encryptedBarcode . '/' . $location;
-        $html2 = '<br><a href="' . $link . '" target="_blank">' . $link  . '</a>';
-
-        $returnval .= $html . $html2;
-    }
-    
-    return [
-      '#title' => 'pickup request test',
-      '#markup' => $returnval
-    ];
+          if (strlen($requestId) > 0) {
+              $encryptedRequestId = md5($pickup_requests_salt . $requestId);
+              $returnval = '<p> Encrypting RequestId: ' . $requestId . ' -> ' . $encryptedRequestId . '<br>';
+              $returnval .= 'pickup_requests_salt: ' . $pickup_requests_salt . '</p><br>';
+          }
+          else {
+              if (strlen($patronId) > 0) {
+                  $barcode =  $this->barcodeFromPatronId($patronId);
+              } else {
+                  $patronId = $this->patronIdFromBarcode($barcode);
+              }
+              if (14 === strlen($barcode)) {
+              if (strlen($row) > 0) {
+                      $barcode = $row;
+                  }
+              $encryptedBarcode = md5($pickup_requests_salt . $barcode);
+                  $returnval = '<h2>' . $patronId .' -> '. $barcode . ' -> ' . $encryptedBarcode . '</h2><br>';
+              
+                  $host = 'http://nginx.docker.localhost:8000';
+                  $link = $host . '/pickuprequest/' . $patronId . '/'. $encryptedBarcode . '/' . $location;
+                  $html2 = '<br><a href="' . $link . '" target="_blank">' . $link  . '</a>';
+
+                  $returnval .= $html2;
+              }
+          } 
+      }
+      return [
+        '#title' => 'pickup request test',
+        '#markup' => $returnval
+      ];
   }
 
   public function pickup_request($pnum, $encrypted_barcode, $loc) {
-    $mode = \Drupal::request()->query->get('mode');
-    $requestPickup_html = '';
-    if ($this->validateTransaction($pnum, $encrypted_barcode)) {
-        $requestPickup_html = \Drupal::formBuilder()->getForm('Drupal\arborcat\Form\UserPickupRequestForm', $pnum, $loc, $mode);
-    } else {
-        drupal_set_message('The Pickup Request could not be processed');
-    }
-    $render[] = [
-            '#theme' => 'pickup_request_form',
-            '#formhtml' => $requestPickup_html,
-            '#max_locker_items_check' => \Drupal::config('arborcat.settings')->get('max_locker_items_check')
-        ];
-    return $render;
+      $mode = \Drupal::request()->query->get('mode');
+      $requestPickup_html = '';
+      if ($this->validateTransaction($pnum, $encrypted_barcode)) {
+          $requestPickup_html = \Drupal::formBuilder()->getForm('Drupal\arborcat\Form\UserPickupRequestForm', $pnum, $loc, $mode);
+      } else {
+          drupal_set_message('The Pickup Request could not be processed');
+      }
+      $render[] = [
+              '#theme' => 'pickup_request_form',
+              '#formhtml' => $requestPickup_html,
+          ];
```

Comment:
Do we need the #max_locker_items_check put back into the render array?

Reply:
Yes it does need to be passed in the render array to the userpickuprequests form.
When I merged the prodtest branch into pickup-requests-cancel branch - I had some conflicts and it appears that the addition to the renderarray (in prodtest) was removed during the merge to pickup-requests-cancel.

Reply:
added missing max_locker_items_check renderArray element.